### PR TITLE
Adopt metav1 condition

### PR DESCRIPTION
**Pull request by qiujian16**
·

### CODE REVIEW

File: cluster/v1/types.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ type ClientConfig struct {
 // ManagedClusterStatus represents the current status of joined managed cluster.
 type ManagedClusterStatus struct {
 	// Conditions contains the different condition statuses for this managed cluster.
-	Conditions []StatusCondition `json:"conditions"`
+	Conditions []metav1.Condition `json:"conditions"`
```
Comment (deads2k) [inline cluster/v1/types.go:87]: be sure your current data is compatible with this change

**Comment by deads2k** [comment]
Be sure any existing data you write is compatible with this change, but I'm in favor and I think it will make your life easier in the future.

/approve

File: cluster/v1/types.go (hunk 1 above)
Comment (qiujian16) [reply cluster/v1/types.go:87]: yes, i am quite sure it is compatible.

**Pull request merged**